docker-java-api/src/main/java/com/github/dockerjava/api/model/ConfigSpec.java

```diff
@@ -0,0 +1,83 @@
+package com.github.dockerjava.api.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import javax.annotation.CheckForNull;
+import java.io.Serializable;
+import java.util.Base64;
+import java.util.Map;
+
+/**
+ * @since {@link RemoteApiVersion#VERSION_1_30}
+ */
+@EqualsAndHashCode
+@ToString
+public class ConfigSpec implements Serializable {
+    public static final long serialVersionUID = 1L;
+
+    /**
+     * @since 1.30
+     */
+    @JsonProperty("Name")
+    private String name;
+
+    /**
+     * @since 1.30
+     */
+    @JsonProperty("Data")
+    private String data;
+
+    /**
+     * @since 1.30
+     */
+    @JsonProperty("Labels")
+    private Map<String, String> labels;
+
+    /**
+     * @see #name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @see #name
+     */
+    public ConfigSpec withName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * @see #data
+     */
+    public byte[] getData() {
+        return Base64.getDecoder().decode(data);
+    }
+
+    /**
+     * @see #data
+     */
+    public ConfigSpec withData(byte[] data) {
+        this.data = Base64.getEncoder().encodeToString(data);
```

Comment:
This should really use 
```
Base64.getUrlEncoder().encodeToString(data)
```
The specs specify that it is URL safe

Reply:
Can you add the following

    /**
     * This gets the decoded data as UTF-8 string.
     *
     * @return decoded data
     */
    public String getDataAsString() {
        return new String(getData(), StandardCharsets.UTF_8);
    }

I think most use cases would store the config data as text.